src/hidimstat/desparsified_lasso.py

```diff
@@ -152,271 +122,382 @@ def desparsified_lasso(
 
     References
     ----------
-    .. [1] Zhang, C. H., & Zhang, S. S. (2014). Confidence intervals for
-           low dimensional parameters in high dimensional linear models.
-           Journal of the Royal Statistical Society: Series B: Statistical
-           Methodology, 217-242.
-
-    .. [2] Van de Geer, S., Bühlmann, P., Ritov, Y. A., & Dezeure, R. (2014).
-           On asymptotically optimal confidence regions and tests for
-           high-dimensional models. Annals of Statistics, 42(3), 1166-1202.
-
-    .. [3] Javanmard, A., & Montanari, A. (2014). Confidence intervals and
-           hypothesis testing for high-dimensional regression. The Journal
-           of Machine Learning Research, 15(1), 2869-2909.
-
-    .. [4] Bellec, P. C., & Zhang, C. H. (2019). De-biasing the lasso with
-           degrees-of-freedom adjustment. arXiv preprint arXiv:1902.08885.
-
-    .. [5] Celentano, M., Montanari, A., & Wei, Y. (2020). The Lasso with
-           general Gaussian designs with applications to hypothesis testing.
-           arXiv preprint arXiv:2007.13716.
+    .. footbibliography::
     """
 
-    X = np.asarray(X)
-
-    n_samples, n_features = X.shape
-
-    memory = check_memory(memory)
+    X_ = np.asarray(X)
+
+    n_samples, n_features = X_.shape
+    if group:
+        n_times = y.shape[1]
+        if cov is not None and cov.shape != (n_times, n_times):
+            raise ValueError(
+                f'Shape of "cov" should be ({n_times}, {n_times}),'
+                + f' the shape of "cov" was ({cov.shape}) instead'
+            )
+
+    # centering the data and the target variable
+    y_ = y - np.mean(y)
+    X_ = X_ - np.mean(X_, axis=0)
+
+    # Lasso regression and noise standard deviation estimation
+    # TODO: other estimation of the noise standard deviation?
```

Comment:
1. Do we want other functions for the estimation of the standard deviation of the noise?

Reply:
No.